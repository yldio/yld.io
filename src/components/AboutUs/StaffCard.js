import React from 'react'
import styled from 'styled-components'
import remcalc from 'remcalc'
import { Col } from '../grid'
import { Padding } from 'styled-components-spacing'
import { BodyPrimary, Subtitle } from '../Typography'

import ExternalAnchor from '../Common/ExternalAnchor'
import Image from '../Common/Image'

const IMAGE_SIZE = 24

const TappableAnchor = styled(ExternalAnchor)`
  width: ${props => remcalc(props.theme.elementSizes.tappableArea)};
  height: ${props => remcalc(props.theme.elementSizes.tappableArea)};
  display: flex;
  justify-content: center;
  align-items: center;

  &:first-child {
    margin-left: -${props => remcalc((props.theme.elementSizes.tappableArea - IMAGE_SIZE) / 2)};
  }
`

const LinksContainer = styled.div`
  display: flex;
  margin-top: -${props => remcalc((props.theme.elementSizes.tappableArea - IMAGE_SIZE) / 2)};
  padding-bottom: ${props =>
    remcalc(18 - (props.theme.elementSizes.tappableArea - IMAGE_SIZE) / 2)};
`

const Description = styled(BodyPrimary)`
  padding-bottom: ${props => remcalc(30)};
`

const SocialLink = ({ name, url, image }) => {
  const size = `${IMAGE_SIZE}px`

  return (
    <TappableAnchor href={url}>
      <Image image={image} alt={name} width={size} height={size} />
    </TappableAnchor>
  )
}

const SocialLinks = ({ data }) => {
  return (data || []).length > 0 ? (
    <LinksContainer>
      {data.map(({ name, url, image }, idx) => (
        <SocialLink
          key={`${name}-${idx}`}
          name={name}
          url={url}
          image={image}
        />
      ))}
    </LinksContainer>
  ) : null
}

const StaffCard = ({ name, image, role, description, socialLinks = [] }) => {
  return (
    <Col width={[1, 1, 1, 1, 6 / 12, 4 / 12]}>
      <Padding bottom={{ smallPhone: 3, smallTablet: 3.5, tablet: 5 }}>
        <Padding bottom={2}>
          <Image image={image} width="100%" />
        </Padding>
        <Subtitle noPadding>{name}</Subtitle>
        <Padding bottom={3}>
          <Subtitle noPadding muted>
            {role}
          </Subtitle>
        </Padding>
        <Description noPadding>{description}</Description>
        <SocialLinks data={socialLinks} />
      </Padding>
    </Col>
  )
}

export default StaffCard
