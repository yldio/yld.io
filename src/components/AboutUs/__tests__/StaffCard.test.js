import React from 'react'
import renderComponent from '../../../utils/tests/renderComponent'
import StaffCard from '../StaffCard'

describe('StaffCard', () => {
  it('renders correctly', () => {
    const details = {
      name: 'Nuno Job',
      image: {
        file: {
          url: 'img1.jpg'
        }
      },
      role: 'Co-founder & Chief Executive Officer - YLD',
      description:
        "Nuno is the Co-founder and CEO of YLD Group. Previously he was the Chief Commercial Officer at Nodejitsu, where he was responsible for the world's largest Node.js cloud, providing extensive contributions to the success of Node.js. Nuno's formative work years were spent in the USA at IBM Research and MarkLogic. He is also a proud Stanford alumni.",
      socialLinks: [
        {
          image: {
            title: 'LinkedIn Icon',
            file: {
              url:
                '//images.ctfassets.net/22g1lenhck4z/3uuYxyNek0bMtaGIwoXFei/586067fb0cf887cbf18bcb12588bda71/linkedin-icon.svg'
            },
            fluid: {}
          },
          name: 'Nuno Job Linkedin',
          url: 'https://www.linkedin.com/in/nunojob/'
        }
      ]
    }
    const tree = renderComponent(<StaffCard {...details} />)

    expect(tree).toMatchSnapshot()
  })
})
