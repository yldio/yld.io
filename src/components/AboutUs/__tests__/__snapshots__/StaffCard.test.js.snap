// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StaffCard renders correctly 1`] = `
<div
  className="sc-bwzfXH llXZFL"
  width={
    Array [
      1,
      1,
      1,
      1,
      0.5,
      0.3333333333333333,
    ]
  }
>
  <div
    className="sc-ifAKCX glTJYn"
  >
    <div
      className="sc-ifAKCX iSGemK"
    >
      <img
        className="sc-gZMcBi dhwgne"
        src="img1.jpg"
        width="100%"
      />
    </div>
    <h3
      className="sc-htoDjs hHzTme"
    >
      Nuno Job
    </h3>
    <div
      className="sc-ifAKCX cCGtBF"
    >
      <h3
        className="sc-htoDjs lnaRzG"
        muted={true}
      >
        Co-founder & Chief Executive Officer - YLD
      </h3>
    </div>
    <p
      className="sc-jTzLTM hhhpkm"
    >
      Nuno is the Co-founder and CEO of YLD Group. Previously he was the Chief Commercial Officer at Nodejitsu, where he was responsible for the world's largest Node.js cloud, providing extensive contributions to the success of Node.js. Nuno's formative work years were spent in the USA at IBM Research and MarkLogic. He is also a proud Stanford alumni.
    </p>
    <div
      className="sc-VigVT fkYcKR"
    >
      <a
        className="sc-gqjmRU kqkEpH"
        href="https://www.linkedin.com/in/nunojob/"
        onClick={[Function]}
        rel="noopener noreferrer"
        target="_blank"
        title="Nuno Job Linkedin account"
      >
        <img
          alt="Nuno Job Linkedin"
          className="sc-gZMcBi gUkkRN"
          height="24px"
          src="//images.ctfassets.net/22g1lenhck4z/3uuYxyNek0bMtaGIwoXFei/586067fb0cf887cbf18bcb12588bda71/linkedin-icon.svg"
          width="24px"
        />
      </a>
    </div>
  </div>
</div>
`;
